### PR TITLE
Generated rc

### PR DESCRIPTION
**Pull request by sithlord48**
## Contributor Checklist:

* [ ] This is a user-visible change and I have created a file in the `doc/newsfragments` directory (and made sure to read the `README.md` in that directory)
* [X] This is not a user-visible change

based on #1946 
  - Use Cmake to generate version info in rc files reducing the need to have as many defines

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,12 @@ option(INPUTLEAP_USE_EXTERNAL_GTEST "Use external installation of Google Test fr
 option(INPUTLEAP_BUILD_X11 "Build with XWindows support" ON)
 option(INPUTLEAP_BUILD_LIBEI "Build with libei support" OFF)
 option(INPUTLEAP_DEPLOY_FLATPAK_SCRIPT "Deploy Flatpak script [Experimental]" OFF)
+
+include (cmake/Version.cmake)
```
Comment (p12tic) [inline CMakeLists.txt:29]: Please make spacing consistent at least in lines that this PR touches.

**Review by p12tic — CHANGES_REQUESTED** [review]
LGTM, just one nit.

File: CMakeLists.txt (hunk 1 above)
Comment (sithlord48) [reply CMakeLists.txt]: Fixed in #1946 (that should land first since its this items base

**Review by p12tic — CHANGES_REQUESTED** [review]
Looking through the PR a second time, I don't agree with the approach.

The version information should come two ways:

- If it's release, then hardcoded INPUTLEAP_VERSION_* variables. Probably add a INPUTLEAP_IS_RELEASE flag to select this path.

- If it's not release (default), use whatever git supplies.

Overall most the PR is fine, this is just relatively small policy decision.

**Comment by sithlord48** [comment]
> Looking through the PR a second time, I don't agree with the approach.
> 
> The version information should come two ways:
> 
>     * If it's release, then hardcoded INPUTLEAP_VERSION_* variables. Probably add a INPUTLEAP_IS_RELEASE flag to select this path.
> 
>     * If it's not release (default), use whatever git supplies.
> 
> 
> Overall most the PR is fine, this is just relatively small policy decision.

Why over complicate this with another yet another flag? If its a release then the hard code and git tag will match.

**Comment by shymega** [comment]
I agree here with @sithlord48, FWIW.

**Comment by p12tic** [comment]
I believe rc file generation could be submitted as a separate PR, no? https://github.com/input-leap/input-leap/pull/1946 is not going to be merged as is.

**Comment by sithlord48** [comment]
Sure you do what ever you want with it

**Pull request closed without merging**